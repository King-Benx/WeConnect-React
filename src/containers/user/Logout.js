import React from 'react';
import { Redirect } from 'react-router-dom';
import axios from 'axios';
import { BASE_URL } from '../../custom/constants';
import CustomFunctions from '../../custom/CustomFunctions';

class Logout extends React.Component{
 // logouts the user and blacklists a token from the api side at the same time
    render(){
        axios.post(BASE_URL+'api/v1/auth/logout',{},
    {headers:{'x-access-token':CustomFunctions.getToken()}})
    .then(res=>{
        CustomFunctions.createNotifications(res.status,res.data.message);
        CustomFunctions.deleteToken();
    })
    .catch(err => { 
        CustomFunctions.createNotifications(err.status, err.response.data.message);   
    });
    return <Redirect to={{pathname:'/'}}/>
    }
}
export default Logout;