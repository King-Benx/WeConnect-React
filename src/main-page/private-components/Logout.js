import React from 'react';
import {Redirect} from 'react-router-dom';
import superagent from 'superagent';
import { BASE_URL } from '../functional-resources/urls';

const Logout = ()=>{
    superagent.post(BASE_URL+'api/v1/auth/logout').send({}).set({'x-access-token':localStorage.getItem('token')}).end((err,res)=>{
        if(err){alert(res.body.message)}; 
        alert(res.body.message);
        return <Redirect to="/"/>
    });
    localStorage.removeItem('token');

    return (
        <Redirect to="/"/>
    );
}
export default Logout;