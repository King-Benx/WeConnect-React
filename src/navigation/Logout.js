import React from 'react';
import {Link} from 'react-router-dom';

const Logout = ()=>(
    <li>
        <Link to='/logout'>
            <b><i className="glyphicon glyphicon-log-out"></i> Logout</b>
        </Link>  
    </li>
)
export default Logout;