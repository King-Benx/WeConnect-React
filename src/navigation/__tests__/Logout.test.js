import React from 'react';
import { shallow } from 'enzyme';
import Logout from "../Logout";
import renderer from 'react-test-renderer';

describe('These are tests the logout link', () => {
    const event={
        target:{
            value:{}
        },
        preventDefault: () => {

        }
    }

    let mountedLogoutLink;

    beforeEach(()=>{
        mountedLogoutLink= shallow(<Logout />);
    });

    it('renders without crashing', () => {
        mountedLogoutLink =  shallow(<Logout />);
        const tree = renderer.create(<Logout />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('has links', () =>{
        const links = mountedLogoutLink.find('Link')
        expect(links.length).toBe(1)
    });

    it('has a list', () =>{
        const lists = mountedLogoutLink.find('li')
        expect(lists.length).toBe(1)
    });

});