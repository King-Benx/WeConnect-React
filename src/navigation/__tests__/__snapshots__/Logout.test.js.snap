// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`These are tests the logout link renders without crashing 1`] = `
<li>
  <a
    href="/logout"
    onClick={[Function]}
  >
    <b>
      <i
        className="glyphicon glyphicon-log-out"
      />
       Logout
    </b>
  </a>
</li>
`;
